### ISSUE TITLE
Can I customize it if I want to have so many features in calendar

### ISSUE THREAD

**Issue by ankitoodles** [angular-ui/ui-calendar#208]
I want to develop a calendar in which user can have much options for marking his availability,non availability,events,others availability,birthdays,meetings,venue details and so many things.

I want to develop it in angularjs.

I have two options :

 1.   Use angular ui calendar and customize html and js files.

    Here I think it will consume so much efforts in customizing the calendar and at some time I may stuck and I will not be available to add more features.

 2.    Develop my own calendar and implement all what I want.

    Here the only problem is I am not sure about how complex the development of calendar is,Because I have to write whole logic like time zone handling,day light savings etc.

So which option is better?

The main thing is calendar will be too complex and flexible.

**Comment by hanifbbz** [comment]
I had a similar assignment for a proprietary project. We started with Fullcalendar, then switched to Angular UI Calendar (which is buggy and has a weird interface), so we ended up making our own calendar which adds the missing features and we have full control over its look and feel. Our calendar took 5 days to develop and has roughly 160 lines of code and about 80 lines of CSS. It uses moment.js and Angular. That's all. I would definitely recommend option 2.

**Comment by thathoo** [comment]
@hanifbbz thanks for your comment here, very helpful. Basically you are suggesting not to use Angular UI Calendar for a big production project? And build our own? You said you were able to finish the calendar in 5 days, did you use angular for it? Were there any other prices of open source libraries you used? We are in a similar position as @ankitoodles and need to make a decision quickly. Thanks!

**Comment by hanifbbz** [comment]
@thathoo yeah we used just Angular. I think angular-ui project is doing its best to wrap the fullcalendar, but it's just not enough. We had some trouble working with the event model and customizing it beyound CSS styles. It's a very good product for prototyping but if you're working on a multi-million dollar project, you probably want to have control over your source code and customize it.

TL;DR: if you have the resources to implement a calendar, do it.

**Comment by joshkurz** [comment]
@hanifbbz dogging on the calendar, Shame on you. :) But you bring up good points. I really think the future is going to be some type of web component based calendar, which is more plug and play vs such a huge monolithic library.
